### ISSUE TITLE
Compiler error on type query

### ISSUE THREAD

**Issue by vinodgro** [google-research/dex-lang#390]
>=> :t \x y. %fadd x y
Compiler bug!
Please report this at github.com/google-research/dex-lang/issues
Type error:Expected a fixed-width floating-point type, but found: ?
Querying:
(%fadd x y)
CallStack (from HasCallStack):
  error, called at src/lib/PPrint.hs:738:25 in dex-0.1.0.0-Gdb5EIM401y5JLGj9rRxaQ:PPrint
  ignoreExcept, called at src/lib/Type.hs:48:13 in dex-0.1.0.0-Gdb5EIM401y5JLGj9rRxaQ:Type
  getType, called at src/lib/Embed.hs:94:25 in dex-0.1.0.0-Gdb5EIM401y5JLGj9rRxaQ:Embed

**Comment by apaszke** [comment]
Yeah this is not very surprising. The `%XYZ` compiler builtins do not necessarily have all the inference rules implemented (+ there's a restriction that they can only take variables as arguments). This is also why having them in a position with an ambiguous type is an error that you see. You can see that in the prelude, all lambdas that use those builtins have their arguments annotated with a type.

**tl;dr** %-functions are internal and might have a ton of rough edges. But you shouldn't need to use them in user code! All of them (well, with the exception of `%inject`) are wrapped in the prelude already!

**Comment by vinodgro** [comment]
sounds good. maybe you should change the message. The message asks  to file an issue. ;)

**Comment by apaszke** [comment]
Good point 😆 In that case thanks for opening the issue!
